File: Data Structures & Algorithms/two-integer-sum/submission-2.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {

        val tracker = HashMap<Int, Int>();
        var index = 0;
        for(n in nums) {
            if(tracker.containsKey(target - n)) return intArrayOf(tracker.get(target - n)!!, index);
            tracker.put(n, index);
            index++;
        }
        return intArrayOf()
    }
}

File: Data Structures & Algorithms/binary-search/submission-2.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0, r = nums.size() - 1;
        int mid = l + (r - l) / 2;
        while(l <= r) {
            if(nums[mid] < target) {
                l = mid + 1;
                mid = l + (r - l) / 2;
            }
            else if(nums[mid] > target) {
                r = mid - 1;
                mid = l + (r - l) / 2;
            }
            else if (nums[mid] == target) {
                return mid;
            }
        }
        return -1;
    }
};
